web/app/js/components/BreadcrumbHeader.jsx

```diff
@@ -73,27 +73,26 @@ class BreadcrumbHeader extends React.Component {
     }
   }
 
-  renderBreadcrumbSegment(segment, shouldPluralizeFirstSegment) {
+  renderBreadcrumbSegment(segment, numCrumbs, index) {
     let isMeshResource = isResource(segment);
 
     if (isMeshResource) {
-      if (!shouldPluralizeFirstSegment) {
-        return friendlyTitle(segment).singular;
+      if (numCrumbs === 1 || index !== 0) {
```

Comment:
it took me a little while to go through the logic for this (there's a lot going on in these functions!) mind adding a helper comment?
```suggestion
      if (numCrumbs === 1 || index !== 0) {
        // pluralize the noun for the list pages ("Namespaces", as well as
        // "Namespace > linkerd > Deployments")
```

Reply:
Yes, I agree that the logic is a bit confusing, since our pluralization is attempting to mirror human language which is inconsistent. Will add clarifying note.